lupa/_lupa.pyx

```diff
@@ -1399,12 +1399,22 @@ cdef bint call_python(LuaRuntime runtime, lua_State *L, py_object* py_obj) excep
 
 cdef int py_call_with_gil(lua_State* L, py_object *py_obj) with gil:
     cdef LuaRuntime runtime = None
+    cdef lua_State* stored_state
```

Comment:
I would just initialise this to NULL and store it back if it's not NULL. There's no need for an additional flag.